chips/sam4l/src/eic.rs

```diff
@@ -0,0 +1,377 @@
+//! Implementation of the SAM4L External Interrupt Controller (EIC).
+//!
+//! Datasheet section "21. External Interrupt Controller (EIC)".
+//!
+//! The External Interrupt Controller (EIC) allows pins to be configured as external
+//! interrupts. Each external interrupt has its own interrupt request and can be individually
+//! interrupt_lineed. Each external interrupt can generate an interrupt on rising or falling edge, or
+//! high or low level. Every interrupt input has a configurable filter to remove spikes from
+//! the interrupt source. Every interrupt pin can also be configured to be asynchronous in order
+//! to wake up the part from sleep modes where the CLK_SYNC clock has been disabled.
+//!
+//! - Author: Josh Zhang  <jiashuoz@cs.princeton.edu>
+//! - Updated: June 25, 2019
+
+use crate::pm::{self, Clock, PBDClock};
+use core::cell::Cell;
+use kernel::common::cells::OptionalCell;
+use kernel::common::registers::{register_bitfields, ReadOnly, ReadWrite, WriteOnly};
+use kernel::common::StaticRef;
+use kernel::hil;
+
+/// Representation of an EIC line on the SAM4L.
+pub struct EicLine {
+    line_number: u32,
+}
+
+#[derive(Copy, Clone, Debug)]
+#[repr(u8)]
+enum Line {
```

Comment:
You are definitively on right path but please use `CamelCase` for enum variants,

I would prefer something like:
```rustc
enum Line {
    Nmi = 1
    Ext1 = 2,
    Ext2 = 4,
    ....
    Ext8 = 256,
}
```
Then we could cast it directly to an u32 and don't need to shift when configuring interrupt(s).

Reply:
Alternatively, explicit matching on the enum and configure the bit accordingly.